### DISCUSSION TITLE
call, apply 메서드의 활용방법에는 무엇이 있나요?

### DISCUSSION THREAD

**Discussion by pipisebastian** [publdaze/js-study#108] [this]
- call, apply 메서드가 this 바인딩 이외에는 어떻게 사용될 수 있나요?

**Comment by publdaze** [comment]
> * call, apply 메서드가 (🚧this 바인딩 이외)에는 어떻게 사용될 수 있나요?

- this 바인딩 이외의 기능이라고 할 수 있을 지는 모르겠지만,
- 유사 배열 객체를 배열처럼 다룰 때 사용 가능
- slice, map 등과 같은 배열 메서드는 this 사용하며 동작
- 따라서 call, apply 메서드를 통해 유사 배열 객체를 this로 명시적 바인딩해주면
- 배열 배서드는 유사 배열 객체를 배열처럼 처리 가능

**Comment by pipisebastian** [comment]
> call, apply 메서드의 활용방법에는 무엇이 있나요?

**유사배열객체에, 배열 메서드를 사용하고 싶을때!**

- 함수의 arguments 객체는 유사 배열 객체로 배열 메서드를 사용하기 때문에 slice와 같은, 배열 메서드를 사용하고 싶을때 사용합니다.

**유사배열객체의 특징은?**

- 인덱스로 접근 가능
- length 속성
- 배열 메서드 사용 불가

**유사배열객체를 배열로 만드는 다른 방법은?**

- call, apply 메서드 사용
- Array.from
- 스프레드 문법

**Reply by pipisebastian** [reply]
사실 이거 원리 제대로 이해 못함! 공부필요!!
